### DISCUSSION TITLE
Proposal: use [x,y] indicate/define a array, like json

### DISCUSSION THREAD

**Discussion by ahdung** [dotnet/csharplang#1968] [Language Ideas]
for example:
`var arr = [3, 5];`
equals:
`var arr = new[] {3, 5};`

and not only for initialize, i hope can do this:
```
[3] is Array; // true
[3] is int[]; //true
[3].Length == 1; //true
[] is object[]; //true
[].Length == 0; //true
string.Join(",", ["a", "b"]);
```

**Comment by YairHalberstadt** [comment]
Would you be allowed to specify a type beforehand? That could lead to ambiguities.

Eg.

```csharp
var Int32 = new []{0, 1};
var x = Int32[0]; // indexer, or new array?
```

**Comment by ahdung** [comment]
@YairHalberstadt ofcourse it's should a indexer in this scene, I think it can be distinguished.

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/1238

**Comment by Austin-bryan** [comment]
`[3].Length == 1; //true`

Wouldn't that be false and the length been 3?

**Comment by ahdung** [comment]
@willard720 [x,y,z].Length == 3, [x] indicate a array has only one element.
